BrofilerApp/ViewModels/SummaryViewerModel.cs

```diff
@@ -184,10 +194,12 @@ public ICommand SaveCurrentAttachmentCommand
                   {
                       try
                       {
-                          string defaultExt =
-                            (CurrentAttachment.FileType == SummaryPack.Attachment.Type.BRO_IMAGE) ? "png" : "txt";
-
-                          if (_dialogService.SaveFileDialog(CurrentAttachment.Name, defaultExt) == true)
+                          string defaultExt = Path.GetExtension(CurrentAttachment.Name);
+                          string filter = (CurrentAttachment.FileType == SummaryPack.Attachment.Type.BRO_IMAGE) ?
+                          "Image Files (*.png, *.jpeg, *.bmp, *.gif)|*.png;*.jpg;*.bmp;*.gif|All files (*.*)|*.*" :
```

Comment:
Filter should be `String.Format("(*.{0})|*.{0}", defaultExt)`. No need to list all the possible extensions - we can't convert file type anyway, so we need just to keep the original extension for the file and use it as a filter.